sopel/loader.py

```diff
@@ -104,10 +104,35 @@ def clean_callable(func, config):
 
 
 def is_triggerable(obj):
-    return any(
-        hasattr(obj, attr)
-        for attr in ('rule', 'event', 'intents', 'commands',
-                     'nickname_commands'))
+    """Check if ``obj`` can handle the bot's triggers.
+
+    :param obj: any :term:`function` to check.
+    :return: ``True`` if ``obj`` can handle the bot's triggers.
+
+    A triggerable is a callable that will be used by the bot to handle a
+    particular trigger (i.e. an IRC message): it can be a regex rule, an event,
+    an intent, a command or nickname command. However, it must not be a job
+    or an URL callback.
+
+    .. seealso::
+
+        The :mod:`sopel.module` defines decorators to make a function a
+        triggerable object.
+    """
+    forbidden = (
+        'interval',
+        'url_regex',
+    )
+    must_not = not any(hasattr(obj, attr) for attr in forbidden)
+
+    allowed = (
+        'rule',
+        'event',
+        'intents',
+        'commands',
+        'nickname_commands',
+    )
+    return must_not and any(hasattr(obj, attr) for attr in allowed)
```

Comment:
I think it would be clearer if the above could be refactored so `forbidden` and `allowed` were the names of variables used in the return here, so this could become:

```suggestion
    return allowed and not forbidden
```

Mostly, I just don't like how the var name `must_not` reads here. 😁

Reply:
I'll find a way!